manifests/database.pp

```diff
@@ -2,6 +2,7 @@
 # @api private
 class pulpcore::database (
   Integer[0] $timeout = 3600,
+  Boolean $always_run_migrations = true,
```

Comment:
[question] should this default to `false` (idempotent by default for people using the puppet module directly), with hiera in foreman-installer providing a `true` value for our purposes?

as an added benefit, the acceptance test here wouldn't rely on a non-default value... but is there a major downside that I'm overlooking?

Reply:
That's option 1 from https://github.com/theforeman/puppet-pulpcore/pull/351#pullrequestreview-2238269017 and yes we can certainly make that change.

The thing is, that will make the *default* installation by definition broken.

Reply:
ah, I missed that review comment... thanks!

> The thing is, that will make the _default_ installation by definition broken.

since this is the first time the issue has occurred, it seems reasonable to me to assume that the types of changes requiring a --migrate run despite no pending migrations would be infrequent.

it also occurs to me now that the pulp folks could accompany such future changes with an empty dummy migration if we ask nicely... in theory, I think we could even submit such a patch to pulp and then simply revert this one here -- what do you think?

Reply:
ah nevermind my idea about dummy migrations, I see the comment about https://github.com/pulp/pulpcore/pull/952 now

Reply:
> since this is the first time the issue has occurred, it seems reasonable to me to assume that the types of changes requiring a --migrate run despite no pending migrations would be infrequent.

While I agree such changes are probably infrequent, those are exactly the things you want to have smoothly as otherwise you'll forget or never learn about the special cases before things are broken ;-)

Reply:
I wonder if Pulp could somehow extend https://github.com/django/django/blob/main/django/core/management/commands/migrate.py to just always return true on `--check` instead of lying. But it does not look easily extensible. Hooking into the planner is also something not for the faint of heart.

Reply:
wouldn't it make it non-idempotent again (for Puppet) if `--check` always returns true?